source/backend/api/Services/LeaseService.cs

```diff
@@ -241,6 +249,16 @@ public PimsLease Update(PimsLease lease, IEnumerable<UserOverrideCode> userOverr
             return _leaseRepository.GetNoTracking(lease.LeaseId);
         }
 
+        public IEnumerable<PimsLeaseRenewal> GetRenewalsByLeaseId(long leaseId)
+        {
+            _logger.LogInformation("Getting renewals on lease {leaseId}", leaseId);
+            _user.ThrowIfNotAuthorized(Permissions.LeaseView);
+            var pimsUser = _userRepository.GetByKeycloakUserId(_user.GetUserKey());
+            pimsUser.ThrowInvalidAccessToLeaseFile(_leaseRepository.GetNoTracking(leaseId).RegionCode);
```

Comment:
what does this do for a non-existent lease id?

Reply:
If the lease ID matches nothing it will return an empty collection